File: backend/pm-milestone/src/main/java/com/pm/milestone/domain/Milestone.java
package com.pm.milestone.domain;

import com.baomidou.mybatisplus.annotation.TableName;
import com.pm.common.base.BaseEntity;
import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.time.LocalDate;

@Data
@EqualsAndHashCode(callSuper = true)
@TableName("milestone")
@Schema(description = "里程碑")
public class Milestone extends BaseEntity {

    @Schema(description = "所属项目ID")
    private Long projectId;

    @Schema(description = "里程碑名称")
    private String name;

    @Schema(description = "里程碑说明")
    private String description;

    @Schema(description = "计划日期")
    private LocalDate plannedDate;

    @Schema(description = "实际完成日期")
    private LocalDate actualDate;

    @Schema(description = "状态（pending/in_progress/at_risk/completed/delayed）")
    private String status;

    @Schema(description = "排序序号")
    private Integer sortOrder;
}

File: frontend/src/types/milestone.ts
/** 里程碑状态 */
export type MilestoneStatus = 'pending' | 'in_progress' | 'completed' | 'delayed'

/** 里程碑信息（对齐后端 MilestoneVO） */
export interface Milestone {
  id: number
  projectId: number
  name: string
  description?: string
  plannedDate: string
  actualDate?: string
  status: MilestoneStatus
  sortOrder: number
  createdAt: string
}

/** 里程碑创建请求 */
export interface MilestoneCreateRequest {
  name: string
  description?: string
  plannedDate: string
  sortOrder?: number
}

/** 里程碑更新请求 */
export interface MilestoneUpdateRequest {
  name?: string
  description?: string
  plannedDate?: string
  actualDate?: string
  status?: MilestoneStatus
  sortOrder?: number
}
